src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,108 @@
-export const Pagination = () => {};
+import {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import classNames from 'classnames';
+import { getNumbers } from '../../utils';
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onChange,
+}) => {
+  const [current, setCurrent] = useState<number>(1);
+
+  useEffect(() => {
+    setCurrent(currentPage);
+  }, [currentPage]);
+
+  const pagesAmount = useMemo(
+    () => Math.ceil(total / perPage),
+    [total, perPage],
+  );
+  const pages = useMemo(() => getNumbers(1, pagesAmount), [pagesAmount]);
+
+  const leftButtonDisabled = useMemo(() => current === 1, [current]);
+  const rightButtonDisabled = useMemo(
+    () => current === pagesAmount,
+    [current, pagesAmount],
+  );
+
+  const handleChange = useCallback(() => {
```

Comment:
The handleChange function should not just return prevState, it should return prevState + 1 or prevState - 1 accordingly.